crates/core/src/db/queries/metadatas.rs

```diff
@@ -468,3 +468,55 @@ pub fn activities(
         .load(conn)
         .context("Failed to load nft(s) activities")
 }
+
+const COLLECTION_ACTIVITES_QUERY: &str = r"
+SELECT listings.id as id, metadata, auction_house, price, auction_house, created_at, marketplace_program,
+    array[seller] as wallets,
+    array[twitter_handle_name_services.twitter_handle] as wallet_twitter_handles,
+    'listing' as activity_type
+        FROM listings
+        LEFT JOIN twitter_handle_name_services on (twitter_handle_name_services.wallet_address = listings.seller)
+        WHERE metadata = $1
+    UNION
+    SELECT purchases.id as id, metadata, auction_house, price, auction_house, created_at, marketplace_program,
+    array[seller, buyer] as wallets,
+    array[sth.twitter_handle, bth.twitter_handle] as wallet_twitter_handles,
+    'purchase' as activity_type
+        FROM purchases
+        LEFT JOIN twitter_handle_name_services sth on (sth.wallet_address = purchases.seller)
+        LEFT JOIN twitter_handle_name_services bth on (bth.wallet_address = purchases.buyer)
+        WHERE metadata = $1
+    UNION
+    SELECT offers.id as id, metadata, auction_house, price, auction_house, created_at, marketplace_program,
+    array[buyer] as wallets,
+    array[bth.twitter_handle] as wallet_twitter_handles,
+    'offer' as activity_type
+        FROM offers
+        LEFT JOIN twitter_handle_name_services bth on (bth.wallet_address = offers.buyer)
+        WHERE metadata = $1
+        AND offers.purchase_id IS NULL
+    ORDER BY created_at DESC
+    LIMIT $2
+    OFFSET $3;
+
+ -- $1: address::text
+ -- $2: limit::integer
+ -- $3: offset::integer";
+
+/// Load listing, sales, offers activity for a collection
+///
+/// # Errors
+/// This function fails if the underlying SQL query returns an error
+pub fn collection_activities(
+    conn: &Connection,
+    address: impl ToSql<Text, Pg>,
+    limit: impl ToSql<Integer, Pg>,
+    offset: impl ToSql<Integer, Pg>,
+) -> Result<Vec<NftActivity>> {
+    diesel::sql_query(COLLECTION_ACTIVITES_QUERY)
+        .bind(address)
+        .bind(limit)
+        .bind(offset)
+        .load(conn)
+        .context("Failed to load collection activities")
+}
```

Comment:
Lets move to queries::collection::activites()

The UI also requires some filtering where we need to support include_types: ["LISTINGS", "PURCHASES", "OFFERS"]